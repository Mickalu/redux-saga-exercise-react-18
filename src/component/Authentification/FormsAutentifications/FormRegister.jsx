import React from 'react';

import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Form from "react-bootstrap/Form";
import Button from 'react-bootstrap/Button';

import { changeInputFormIntoState } from '../../../utils/functionGestionInput/gestionFormInput';

const FormRegister = ({ submitRegisterUser, formRegisterValues, setFormRegisterValues, apiResponse }) =>{
  const handleUsernameInput = (inputUsername) => {
    const field = "username";
    changeInputFormIntoState(formRegisterValues, field, inputUsername, setFormRegisterValues);
  };

  const handlePassword = (inputPassword) => {
    const field = "password";
    changeInputFormIntoState(formRegisterValues, field, inputPassword, setFormRegisterValues);
  };

  const handlePassword2 = (inputPassword2) => {
    const field = "password2";
    changeInputFormIntoState(formRegisterValues, field, inputPassword2, setFormRegisterValues);
  }

  const handleEmailInput = (inputEmail) => {
    const field = "email";
    changeInputFormIntoState(formRegisterValues, field, inputEmail, setFormRegisterValues);
  };

  const handleFirstName = (inputFirstName) => {
    const field = "first_name";
    changeInputFormIntoState(formRegisterValues, field, inputFirstName, setFormRegisterValues);
  };

  const handleLastName = (inputLastName) => {
    const field = "last_name";
    changeInputFormIntoState(formRegisterValues, field, inputLastName, setFormRegisterValues);
  };

  const displayErrorMessage = () => {
    if (!apiResponse.status) {
      return (
        <p style={{color: "red"}}>{apiResponse.data}</p>
      );
    }
    else {
      return (null);
    }
  };

  const enableSubmit = () => {
    if (formRegisterValues.password === formRegisterValues.password2 || formRegisterValues.password === "") {
      return false;
    }
    else {
      return true;
    }
  };

  return (
    <Form onSubmit={submitRegisterUser}>
      <Row>
        <Form.Group className='mb-4'>
          <Form.Label>Username : </Form.Label>
          <Form.Control type='text' onChange={(e) => handleUsernameInput(e.target.value)} required/>
        </Form.Group>

        <Form.Group className='mb-4'>
          <Form.Label>Password : </Form.Label>
          <Form.Control data-testid="password" type='password' onChange={(e) => handlePassword(e.target.value)} required/>
        </Form.Group>

        <Form.Group className='mb-4'>
          <Form.Label>Password again : </Form.Label>
          <Form.Control data-testid="password2" type='password' onChange={(e) => handlePassword2(e.target.value)} required/>
        </Form.Group>

        <Form.Group className='mb-4'>
          <Form.Label>Email : </Form.Label>
          <Form.Control type='email' onChange={(e) => handleEmailInput(e.target.value)} required/>
        </Form.Group>

        <Form.Group className='mb-4'>
          <Form.Label>First name: </Form.Label>
          <Form.Control type='text' onChange={(e) => handleFirstName(e.target.value)} required/>
        </Form.Group>

        <Form.Group className='mb-4'>
          <Form.Label>Last name: </Form.Label>
          <Form.Control type='text' onChange={(e) => handleLastName(e.target.value)} required/>
        </Form.Group>
      </Row>

      <Row>
        <Col>
          <Button size="lg" type='submit' disabled={enableSubmit()} data-testid="submit-form-button"> Save </Button>
        </Col>
      </Row>

      <Row>
        {displayErrorMessage()}
      </Row>
    </Form>
  );
};

export default FormRegister
