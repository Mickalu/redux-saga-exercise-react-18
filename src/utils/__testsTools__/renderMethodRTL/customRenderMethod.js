import React from "react";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";

import { store } from "../../../store";

const getByTextCustom = (text) => (
  screen.getByText(text)
);

const queryAllByRoleCustom = (role, options) => (
  screen.queryAllByRole(role, options)
);

const queryByTestIdCustom = (testId) => (
  screen.queryByTestId(testId)
);

const getByTestIdCustom = (testId) => (
  screen.getByTestId(testId)
);

const getByDisplayValueCustom = (displayValue) => (
  screen.getByDisplayValue(displayValue)
);

const renderWithProviders = ({ children }) => (
  <Provider store={store}>
    { children }
  </Provider>
);

export const renderCustom = (ui, options = {}) => render(ui, { wrapper: renderWithProviders, ...options });

// re-export everything
export * from '@testing-library/react';

export { renderCustom as render };
export { getByTextCustom as getByText };
export { queryAllByRoleCustom as queryAllByRole };
export { queryByTestIdCustom as queryBytestId };
export { getByTestIdCustom as getByTestId };
export { getByDisplayValueCustom as getByDisplayValue };
