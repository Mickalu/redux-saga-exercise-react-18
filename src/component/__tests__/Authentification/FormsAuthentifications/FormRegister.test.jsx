import React from "react";

import FormRegister from "../../../Authentification/FormsAutentifications/FormRegister";
import { render, getByTestId, fireEventChange } from "../../../../utils/__testsTools__/renderMethodRTL/customRenderMethod";

const initFormValue = {
  username: "",
  password: "",
  password2: "",
  first_name: "",
  last_name: "",
  email: "",
};

const apiResponse = {
  status: true,
  data: {}
};

it("formRegister should matchsnapshot", () => {
  const { container } = render(
    <FormRegister
      formRegisterValues={initFormValue}
      apiResponse={apiResponse}
    />);

  expect(container).toMatchSnapshot();
});

it("If input password and password2 not same value, button submit disabled", () => {
  const formValueNotSamePassword = {
    username: "",
    password: "password",
    password2: "password2",
    first_name: "",
    last_name: "",
    email: "",
  };

  render(
    <FormRegister
      formRegisterValues={formValueNotSamePassword}
      apiResponse={apiResponse}
    />
  );

  expect(getByTestId("password")).toBeInTheDocument();
  expect(getByTestId("password2")).toBeInTheDocument();
  expect(getByTestId("submit-form-button")).toBeInTheDocument();

  expect(getByTestId("submit-form-button")).toHaveAttribute('disabled')
});

it("Should not dispable submit button if same password value", () => {
  const formValueSamePassword = {
    username: "",
    password: "password",
    password2: "password",
    first_name: "",
    last_name: "",
    email: "",
  };

  render(
    <FormRegister
      formRegisterValues={formValueSamePassword}
      apiResponse={apiResponse}
    />
  );

  expect(getByTestId("submit-form-button")).not.toHaveAttribute('disabled')
});
