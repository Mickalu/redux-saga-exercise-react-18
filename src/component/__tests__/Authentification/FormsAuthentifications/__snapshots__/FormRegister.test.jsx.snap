// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formRegister should matchsnapshot 1`] = `
<div>
  <form
    class=""
  >
    <div
      class="row"
    >
      <div
        class="mb-4"
      >
        <label
          class="form-label"
        >
          Username : 
        </label>
        <input
          class="form-control"
          required=""
          type="text"
        />
      </div>
      <div
        class="mb-4"
      >
        <label
          class="form-label"
        >
          Password : 
        </label>
        <input
          class="form-control"
          data-testid="password"
          required=""
          type="password"
        />
      </div>
      <div
        class="mb-4"
      >
        <label
          class="form-label"
        >
          Password again : 
        </label>
        <input
          class="form-control"
          data-testid="password2"
          required=""
          type="password"
        />
      </div>
      <div
        class="mb-4"
      >
        <label
          class="form-label"
        >
          Email : 
        </label>
        <input
          class="form-control"
          required=""
          type="email"
        />
      </div>
      <div
        class="mb-4"
      >
        <label
          class="form-label"
        >
          First name: 
        </label>
        <input
          class="form-control"
          required=""
          type="text"
        />
      </div>
      <div
        class="mb-4"
      >
        <label
          class="form-label"
        >
          Last name: 
        </label>
        <input
          class="form-control"
          required=""
          type="text"
        />
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <button
          class="btn btn-primary btn-lg"
          data-testid="submit-form-button"
          type="submit"
        >
           Save 
        </button>
      </div>
    </div>
    <div
      class="row"
    />
  </form>
</div>
`;
